Sprint-3/alarmclock/alarmclock.js

```diff
@@ -1,25 +1,78 @@
-function setAlarm() {}
+var audio = new Audio("alarmsound.mp3");
 
-// DO NOT EDIT BELOW HERE
+function setAlarm() {
+  const time = parseInt(document.getElementById("alarmSet").value); // Fetch the value of the input in seconds
 
-var audio = new Audio("alarmsound.mp3");
+  document.querySelector(".quadrat").classList.add("disabled"); // Disable the .quadrat element
 
-function setup() {
-  document.getElementById("set").addEventListener("click", () => {
-    setAlarm();
-  });
+  if (isNaN(time) || time <= 0) {
+    alert("Please enter a valid number of seconds.");
+    return;
+  }
 
-  document.getElementById("stop").addEventListener("click", () => {
-    pauseAlarm();
-  });
+  // Display the initial time remaining
+  document.getElementById("timeRemaining").innerHTML =
+    "Time Remaining: " + formatTime(time);
+
+  // Start the countdown
+  startCountdown(time);
+}
+
+function startCountdown(timeRemaining) {
+  let time = timeRemaining;
+
+  // Update the time remaining every second
+  const countdownInterval = setInterval(() => {
+    if (time <= 0) {
+      clearInterval(countdownInterval);
+      document.getElementById("timeRemaining").innerHTML = "Time's up!";
+      playAlarm(); // Play alarm when time is up
+      triggerFlash(); // Trigger the flashing effect
+    } else {
+      time--; // Decrease time by 1 second
+      document.getElementById("timeRemaining").innerHTML =
+        "Time Remaining: " + formatTime(time);
+    }
+  }, 1000);
 }
 
+// Function to trigger the flashing effect
+function triggerFlash() {
```

Comment:
Test this functionality - does it work? Also, thinking from a accessibility perspective, is this accessible for everyone?